File: Pod/Classes/PURFilter.m
#import "PURFilter.h"
#import "PURLogger.h"
#import "PURLog.h"

@implementation PURFilter

- (instancetype)initWithLogger:(PURLogger *)logger tagPattern:(NSString *)tagPattern
{
    self = [super init];
    _identifier = [NSUUID UUID].UUIDString;
    _tagPattern = tagPattern;
    _logger = logger;

    return self;
}

- (PURLogStore *)logStore
{
    return self.logger.logStore;
}

- (void)configure:(NSDictionary<NSString *, id> *)settings
{

}

- (NSArray<PURLog *> *)logsWithObject:(id)object tag:(NSString *)tag captured:(NSString *)captured
{
    if (![object isKindOfClass:[NSDictionary class]]) {
        return @[];
    }

    NSDate *currentDate = [self.logger currentDate];
    return @[[[PURLog alloc] initWithTag:tag date:currentDate userInfo:object]];
}

@end

File: Pod/Classes/PURFilter.swift
//
//  PURFilter.swift
//  Pods
//
//  Created by admin on 7/27/17.
//
//

import Foundation

class PURFilter {
    private(set) var identifier: String = ""
    private(set) var tagPattern: String?
    private(set) var logger: PURLogger?
    private(set) var logStore: PURLogStore?
    
    init(logger: PURLogger, tagPattern: String?) {
        self.identifier = UUID().uuidString
        self.tagPattern = tagPattern
        self.logger = logger
    }
    
    func getLogStore() -> PURLogStore? {
        return logger?.logStore()
    }
    
    func configure(_ settings: [String: Any]) {
    }
    
    func logs(withObject object: Any, tag: String, captured: String?) -> [PURLog] {
        if !(object is [AnyHashable: Any]) {
            return []
        }
        
        if let currentDate = logger?.currentDate() {
            return [PURLog(tag: tag, date: currentDate, userInfo: object as! [String : Any])]
        }
        
        return []
    }
}
